### ISSUE TITLE
get index of element in array

### ISSUE THREAD

**Issue by mathiaHT** [schibsted/jslt#122]
Hi, I did not find mention of a "indexOf" function. Given an array the function would act like

```
let array = array(["toto", "tutu", "tata"])  
$array.indexOf("toto") // return 0
$array.indexOf("tata") // return 2
```

Is there such function? Thank you!

**Comment by larsga** [comment]
That's a good point. We probably should have that function. You can implement a workaround yourself with

```
def index-of(array, value)
  find-index-of($array, $value, 0)

def find-index-of(array, value, index)
  if ($index >= size($array))
    -1
  else if ($array[$index] == $value)
    $index
  else
    find-array-index($array, $value, $index + 1)

let array = array(["toto", "tutu", "tata"])  
index-of($array, "toto") => 0
index-of($array, "tata") => 2
```

**Comment by mathiaHT** [comment]
Thank you! It makes the job :)

**Comment by larsga** [comment]
Let's not close this, because I think we should add this function.

**Comment by Boy2ang** [comment]
hi eveybody:
i did not find a method to get currentIndex ,  could you help me?

== begin input ==
{
  "list":[
   {"name":"xx1"},  //  index 0
   {"name":"xx2"}   //  index 1
 ]
}
== end output ==

== begin output ==
[for(.list) 

// how i get current index ?
{ "currentIndex": .index}   

] 
==end output ==

**Comment by larsga** [comment]
@Boy2ang This is actually the same as #127 We're actively considering how to solve that one.

**Comment by Boy2ang** [comment]
・       I see. Thank you ! !

发件人: Lars Marius Garshol<mailto:notifications@github.com>
发送时间: 2020年7月9日 2:37
收件人: schibsted/jslt<mailto:jslt@noreply.github.com>
抄送: Yangbo<mailto:ybvip@outlook.com>; Mention<mailto:mention@noreply.github.com>
主题: Re: [schibsted/jslt] get index of element in array (#122)


@Boy2ang<https://github.com/Boy2ang> This is actually the same as #127<https://github.com/schibsted/jslt/issues/127> We're actively considering how to solve that one.

―

**Comment by larsga** [comment]
I'm closing this as a duplicate now.

**Comment by larsga** [comment]
Ooops. That was wrong. The actual issue is something else.

**Comment by larsga** [comment]
The `index-of` function was implemented in commit b8476a8, fixing this issue
